src/jmh/java/org/ajoberstar/gradle/git/release/semver/NearestVersionLocatorBenchmark.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ajoberstar.gradle.git.release.semver;
+
+import org.ajoberstar.grgit.Grgit;
+import org.ajoberstar.grgit.operation.InitOp;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.SecureRandom;
+import java.util.concurrent.TimeUnit;
+
+import static org.ajoberstar.gradle.git.release.semver.NearestVersionLocatorSpec.*;
+
+/**
+ * Benchmarks for {@link NearestVersionLocator}.
+ */
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class NearestVersionLocatorBenchmark {
```

Comment:
Yeah, I'll just cleanup my benchmark/test changes and update the PR branch with the material changes for NearestVersionSpec.